core/app/models/spree/promotion.rb

```diff
@@ -240,8 +240,8 @@ def blacklisted?(promotable)
       case promotable
       when Spree::LineItem
         !promotable.product.promotionable?
-      when Spree::Order
-        promotable.line_items.joins(:product).where(spree_products: { promotionable: false }).exists?
+      # when Spree::Order
+      #   promotable.line_items.joins(:product).where(spree_products: { promotionable: false }).exists?
```

Comment:
Another option could be to perform a check like so:
```
        promotable.line_items.any? &&
          !promotable.line_items.joins(:product).where(spree_products: { promotionable: true }).any?
```

Reply:
Thanks for your suggestion. Do you mean that we could perform this check instead of removing this code?